datafusion/src/execution/context.rs

```diff
@@ -511,9 +515,10 @@ impl ExecutionContext {
     pub async fn write_parquet(
         &self,
         plan: Arc<dyn ExecutionPlan>,
-        path: String,
+        path: impl Into<String>,
         writer_properties: Option<WriterProperties>,
     ) -> Result<()> {
+        let path = path.into();
```

Comment:
This change looks to me like it simply copies its argument an extra time -- in other words `write_parquet` doesn't actually take ownership of the `String` it just needs a reference

If we want to make this more generic (so that it can take anything that can provide itself as a `&str`) perhaps something like:

```rust
  path: impl AsRef<str>
  ...
let path = path.as_ref()
``` 
would be better?

Reply:
yep, good call, i didn't double check whether the logic actually needs the ownership :+1: